ui/job-view/pushes/Push.jsx

```diff
@@ -48,6 +100,132 @@ export default class Push extends React.Component {
     });
   }
 
+  poll() {
+    this.pollIntervalId = setInterval(async () => {
+      const { push } = this.props;
+      const lastModified = this.getLastModifiedJobTime();
+      const jobs = await PushModel.getJobs(push.id, { lastModified });
+
+      this.mapPushJobs(jobs);
+    }, jobPollInterval);
+  }
+
+  toggleSelectedRunnableJob(buildername) {
+    const { selectedRunnableJobs } = this.state;
+    const jobIndex = selectedRunnableJobs.indexOf(buildername);
+
+    if (jobIndex === -1) {
+      selectedRunnableJobs.push(buildername);
+    } else {
+      selectedRunnableJobs.splice(jobIndex, 1);
+    }
+    this.setState({ selectedRunnableJobs: [...selectedRunnableJobs] });
+    return selectedRunnableJobs;
+  }
+
+  async fetchJobs() {
+    const { push } = this.props;
+
+    // if ``nojobs`` is on the query string, then don't load jobs.
+    // this allows someone to more quickly load ranges of revisions
+    // when they don't care about the specific jobs and results.
+    if (getAllUrlParams().has('nojobs')) {
+      return;
+    }
+    const jobs = await PushModel.getJobs(push.id);
+
+    this.mapPushJobs(jobs);
+  }
+
+  mapPushJobs(jobs, skipJobMap) {
+    if (jobs.length > 0) {
+      const { updateJobMap, recalculateUnclassifiedCounts, push } = this.props;
+      const { jobList } = this.state;
+      const newIds = jobs.map(job => job.id);
+      // remove old versions of jobs we just fetched.
+      const existingJobs = jobList.filter(job => !newIds.includes(job.id));
+      const newJobList = [...existingJobs, ...jobs];
+      const sortAndGroupJobs = flowRight(
+        this.sortGroupedJobs,
+        this.groupJobByPlatform,
+      );
+      const platforms = sortAndGroupJobs(newJobList);
```

Comment:
I believe this is equivalent to:
```
const platforms = this.sortGroupedJobs(this.groupJobByPlatform(newJobList));
```

...which would save the need for lodash's `flowRight`?

Reply:
Right again, Professor Moriarty...  er...  Morley...  ;)